Assets/Mirror/Components/NetworkLobbyManager.cs

```diff
@@ -0,0 +1,520 @@
+
+            if (SceneManager.GetActiveScene().name == LobbyScene)
+                RecalculateLobbyPlayerIndices();
+
+            OnLobbyServerDisconnect(conn);
+        }
+
+        public override void OnServerAddPlayer(NetworkConnection conn)
+        {
+            if (SceneManager.GetActiveScene().name != LobbyScene) return;
+
+            if (lobbySlots.Count == maxConnections) return;
+
+            allPlayersReady = false;
+
+            if (LogFilter.Debug) Debug.LogFormat("NetworkLobbyManager:OnServerAddPlayer playerPrefab:{0}", lobbyPlayerPrefab.name);
+
+            GameObject newLobbyGameObject = OnLobbyServerCreateLobbyPlayer(conn);
+            if (newLobbyGameObject == null)
+                newLobbyGameObject = (GameObject)Instantiate(lobbyPlayerPrefab.gameObject, Vector3.zero, Quaternion.identity);
+
+            NetworkLobbyPlayer newLobbyPlayer = newLobbyGameObject.GetComponent<NetworkLobbyPlayer>();
+
+            lobbySlots.Add(newLobbyPlayer);
+
+            RecalculateLobbyPlayerIndices();
+
+            NetworkServer.AddPlayerForConnection(conn, newLobbyGameObject);
+        }
+
+        void RecalculateLobbyPlayerIndices()
+        {
+            if (lobbySlots.Count > 0)
+            {
+                for (int i = 0; i < lobbySlots.Count; i++)
+                {
+                    lobbySlots[i].Index = i;
+                }
+            }
+        }
+
+        public override void ServerChangeScene(string sceneName)
+        {
+            if (sceneName == LobbyScene)
+            {
+                Application.targetFrameRate = 10;
+
+                foreach (NetworkLobbyPlayer lobbyPlayer in lobbySlots)
+                {
+                    if (lobbyPlayer == null) continue;
+
+                    // find the game-player object for this connection, and destroy it
+                    NetworkIdentity uv = lobbyPlayer.GetComponent<NetworkIdentity>();
+
+                    NetworkIdentity playerController = uv.connectionToClient.playerController;
+                    NetworkServer.Destroy(playerController.gameObject);
+
+                    if (NetworkServer.active)
+                    {
+                        // re-add the lobby object
+                        lobbyPlayer.GetComponent<NetworkLobbyPlayer>().ReadyToBegin = false;
+                        NetworkServer.ReplacePlayerForConnection(uv.connectionToClient, lobbyPlayer.gameObject);
+                    }
+                }
+            }
+            else
+            {
+                if (dontDestroyOnLoad)
+                {
+                    foreach (NetworkLobbyPlayer lobbyPlayer in lobbySlots)
+                    {
+                        if (lobbyPlayer != null)
+                        {
+                            lobbyPlayer.transform.SetParent(null);
+                            DontDestroyOnLoad(lobbyPlayer);
+                        }
+                    }
+                }
+
+                Application.targetFrameRate = 60;
```

Comment:
?

Reply:
I think this predated you implementing targetFramerate management, or at least my being aware you had.